data/error_policies/parent_call.cas

```diff
@@ -18,7 +18,29 @@ resource foo inherits bar {
 
 domain dom {
 	foo<qqq>.read(this); // Function does not exist
-	foo<zap>.write(this); // foo does not inherit zap
 	foo<zap>.read(this); // Function does not exist
 	foo<bar>.write(this); // Function does not exist
 }
+
+domain asd {
+	asd<boo>.read_boo_tmp(this); // Function is not castable
+}
+
+virtual resource tmp {
+	@associated_call
+	fn associated_call_from_tmp(domain source) {
+		allow(source, tmp, file, [read]);
+	}
+}
+
+@associate([tmp])
+domain boo {
+	// Creates new resources boo.tmp and implicitly calls
+	// boo.tmp.associated_call_from_tmp(boo)
+	//
+	// boo.tmp inherits tmp
+
+	fn read_boo_tmp(domain source) {
+		allow(source, boo.tmp, file, [read]);
```

Comment:
Okay, I've figure out what I was trying to say in my other comment.  I want an error test where this call is not a built-in.  I'm concerned that that case may not error correctly in the current code.

Reply:
Something like this?
```
domain asd {
	asd<boo>.read_boo_tmp(boo); // Function is not castable
}

virtual resource tmp {
	@associated_call
	fn associated_call_from_tmp(domain source) {
		allow(source, tmp, file, [read]);
	}
}

@associate([tmp])
domain boo {
	// Creates new resources boo.tmp and implicitly calls
	// boo.tmp.associated_call_from_tmp(boo)
	//
	// boo.tmp inherits tmp

	fn read_boo_tmp(domain source) {
		boo.tmp.associated_call_from_tmp(source);
	}
}
```

Reply:
No, I mean passing an associated resource as an argument to a non-builtin.

```
domain asd {
        asd<boo>.some_function(some_resource);
}

virtual resource tmp {
	@associated_call
	fn associated_call_from_tmp(domain source) {
		allow(source, tmp, file, [read]);
	}
}

@associate([tmp])
domain boo {
	// Creates new resources boo.tmp and implicitly calls
	// boo.tmp.associated_call_from_tmp(boo)
	//
	// boo.tmp inherits tmp

	fn read_boo_tmp(domain source) {
		this.some_function(this.tmp);
	}
        fn some_function(resource res) {
                allow(this, res, file, read);
        }
}